src/runtime/pkg/katautils/config.go

```diff
@@ -1228,9 +1244,176 @@ func decodeConfig(configPath string) (tomlConfig, string, error) {
 		return tomlConf, resolved, err
 	}
 
+	err = decodeDropIns(resolved, &tomlConf)
+	if err != nil {
+		return tomlConf, resolved, err
+	}
+
 	return tomlConf, resolved, nil
 }
 
+func decodeDropIns(mainConfigPath string, tomlConf *tomlConfig) error {
+	configDir := filepath.Dir(mainConfigPath)
+	dropInDir := filepath.Join(configDir, "config.d")
+
+	files, err := ioutil.ReadDir(dropInDir)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return fmt.Errorf("error reading '%s' directory: %s\n", dropInDir, err)
```

Comment:
Can you remove all the newlines in these errors and switch `'%s'` to `%q` (equivalent) here and elsewhere:

```suggestion
			return fmt.Errorf("error reading %q directory: %s", dropInDir, err)
```

Reply:
Sure, will do!